docs/connector-development/config-based/declarative-oauth.md

```diff
@@ -0,0 +1,730 @@
+3. `access_token` is located in a users config file at `airbyte.super_secret_credentials.pokemon_access_token`
+
+and we need to make sure that updating the spec to use Declarative OAuth doesn't break existing syncs.
+
+#### Example Declarative OAuth Change
+```diff
+--- a/Users/baz/gl/airbyte/docs/connector-development/config-based/understanding-the-yaml-file/declarative_oauth_examples/base_oauth.yml
++++ b/Users/baz/gl/airbyte/docs/connector-development/config-based/understanding-the-yaml-file/declarative_oauth_examples/legacy_to_decl
+arative_oauth.yml
+@@ -41,9 +41,9 @@ definitions:
+     authenticator:
+       type: OAuthAuthenticator
+       refresh_request_body: {}
+-      client_id: '{{ config["client_id"] }}'
+-      client_secret: '{{ config["client_secret"] }}'
+-      access_token_value: '{{ config["client_access_token"] }}'
++      client_id: '{{ config["super_secret_credentials"]["pokemon_client_id"] }}'
++      client_secret: '{{ config["super_secret_credentials"]["pokemon_client_secret"] }}'
++      access_token_value: '{{ config["super_secret_credentials"]["pokemon_access_token"] }}'
+ 
+ streams:
+   - $ref: "#/definitions/streams/moves"
+@@ -54,68 +54,73 @@ spec:
+     type: object
+     $schema: http://json-schema.org/draft-07/schema#
+     required:
+-      - client_id
+-      - client_secret
+-      - client_access_token
++      - super_secret_credentials
+     properties:
+-      client_id:
+-        type: string
+-        title: Client ID
+-        airbyte_secret: true
+-        order: 0
+-      client_secret:
+-        type: string
+-        title: Client secret
+-        airbyte_secret: true
+-        order: 1
+-      client_access_token:
+-        type: string
+-        title: Access token
+-        airbyte_secret: true
+-        airbyte_hidden: false
+-        order: 2
++      super_secret_credentials:
++        required:
++          - pokemon_client_id
++          - pokemon_client_secret
++          - pokemon_access_token
++        pokemon_client_id:
++          type: string
++          title: Client ID
++          airbyte_secret: true
++          order: 0
++        pokemon_client_secret:
++          type: string
++          title: Client secret
++          airbyte_secret: true
++          order: 1
++        pokemon_access_token:
++          type: string
++          title: Access token
++          airbyte_secret: true
++          airbyte_hidden: false
++          order: 2
+     additionalProperties: true
+   advanced_auth:
+     auth_flow_type: oauth2.0
+     oauth_config_specification:
+       oauth_connector_input_specification:
+         consent_url: >-
+-          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{
+-          redirect_uri_value }}&state={{ state_value }}
++          https://yourconnectorservice.com/oauth/consent?client_id={{client_id_value}}&redirect_uri={{ redirect_uri_value }}&state={{ s
```

Comment:
💅 accidental new line

❓ I dont see what tells our service that the location of `client_id_value` has changed. Shouldnt we have `client_id_key: pokemon_client_id` set somewhere?

Reply:
Yeah! There were missing overrides:
- client_id_key
- client_secret_key

Corrected. Thank you!